src/test/java/com/softserve/controller/LessonsControllerTest.java

```diff
@@ -153,6 +154,95 @@ public void updateLessonIfLessonDoesNotExist() throws Exception {
                 .andExpect(jsonPath("$.group").value(lessonForCompare.getGroup()));
     }
 
+    @Test
+    public void updateForGroupedLesson() throws Exception {
+        LessonInfoDTO lessonDtoForUpdate = new LessonInfoDTO();
+        lessonDtoForUpdate.setId(13L);
+        lessonDtoForUpdate.setHours(2);
+        lessonDtoForUpdate.setLinkToMeeting("https://softserveinc.zoom.us/j/93198369163?pwd=Rk1GU281cDFtK1FCK3pJWXphRkJrQT09");
+        lessonDtoForUpdate.setSubjectForSite("Biology 3");
+        lessonDtoForUpdate.setLessonType(LABORATORY);
+        lessonDtoForUpdate.setTeacher(new TeacherNameMapperImpl().teacherDTOToTeacher(teacherService.getById(5L)));
+        lessonDtoForUpdate.setSubject(new SubjectMapperImpl().subjectToSubjectDTO(subjectService.getById(4L)));
+        lessonDtoForUpdate.setGroup(new GroupMapperImpl().groupToGroupDTO(groupService.getById(4L)));
+        lessonDtoForUpdate.setGrouped(true);
+
+        Lesson expectedLesson = new LessonInfoMapperImpl().lessonInfoDTOToLesson(lessonDtoForUpdate);
+
+        mockMvc.perform(put("/lessons").content(objectMapper.writeValueAsString(lessonDtoForUpdate))
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.id").value(expectedLesson.getId()))
+                .andExpect(jsonPath("$.hours").value(expectedLesson.getHours()))
+                .andExpect(jsonPath("$.linkToMeeting").value(expectedLesson.getLinkToMeeting()))
+                .andExpect(jsonPath("$.subjectForSite").value(expectedLesson.getSubjectForSite()))
+                .andExpect(jsonPath("$.lessonType").value(expectedLesson.getLessonType().toString()))
+                .andExpect(jsonPath("$.subject").value(expectedLesson.getSubject()))
+                .andExpect(jsonPath("$.teacher").value(expectedLesson.getTeacher()))
+                .andExpect(jsonPath("$.group").value(expectedLesson.getGroup()))
+                .andExpect(jsonPath("$.grouped").value(expectedLesson.isGrouped()));
+
+        Lesson GroupedWithSameSubjectForSite = lessonService.getById(14L);
+        Lesson GroupedWithDiffSubjectForSite = lessonService.getById(15L);
+
+        SoftAssertions softAssertions = new SoftAssertions();
+        softAssertions.assertThat(expectedLesson).isEqualToComparingOnlyGivenFields(GroupedWithSameSubjectForSite,
```

Comment:
What are the conditions (parameters) that differ grouped lessons? In other words, what might be different and should be identical about 2 grouped lessons?

Reply:
SubjectForSite parameter